chain/chunks/src/shards_manager_actor.rs

```diff
@@ -297,13 +311,24 @@ impl Handler<ShardsManagerRequestFromClient> for ShardsManagerActor {
     }
 }
 
-impl Handler<ShardsManagerRequestFromNetwork> for ShardsManagerActor {
+impl HandlerWithContext<ShardsManagerRequestFromNetwork> for ShardsManagerActor {
     #[perf]
-    fn handle(&mut self, msg: ShardsManagerRequestFromNetwork) {
-        self.handle_network_request(msg);
+    fn handle(
+        &mut self,
+        msg: ShardsManagerRequestFromNetwork,
+        ctx: &mut dyn DelayedActionRunner<Self>,
+    ) {
+        match self.handle_network_request(msg.clone()) {
+            // Schedule retry processing the message once again if requested
+            HandleNetworkRequestResult::RetryProcessing(dt) => {
+                ctx.run_later("retry processing chunk request", dt, move |this, _ctx| {
+                    let _ = this.handle_network_request(msg);
```

Comment:
```suggestion
                    this.handle_network_request(msg);
```

Isn't this sufficient?